### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by yourivdlans**
This pull updates all the cookbook dependencies, dokku and all its dependencies.

I think its a good idea to go up a major version as we're also updating dokku. This so other people are aware of breaking changes.

Fixes: #19

### CODE REVIEW

**Comment by nickcharlton** [comment]
Thanks for doing this!

Do you think you'd be able to look at those test failures too? I guess it's because it's been quite a while since it all last ran.

**Comment by yourivdlans** [comment]
Yes for sure! Was already having a go at it but it seems some matchers have been removed in newer versions. I need to put some more time in :)

File: spec/unit/recipes/ssh_keys_spec.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
   context "when ssh_keys exist in attributes and no actions are listed" do
     let(:chef_run) do
       runner = ChefSpec::ServerRunner.new do |node|
-        node.set["dokku"]["ssh_keys"] = [{ user: "test", key: "public_key" }]
+        node.override["dokku"]["ssh_keys"] = [{ user: "test", key: "public_key" }]
```
[the remark this answers is not in the document]
Comment (yourivdlans) [reply spec/unit/recipes/ssh_keys_spec.rb]: 😅

File: spec/unit/recipes/ssh_keys_spec.rb [hunk 2]
```diff
@@ -21,7 +21,9 @@
   context "when ssh_keys exist in attributes and no actions are listed" do
     let(:chef_run) do
       runner = ChefSpec::ServerRunner.new do |node|
-        node.set["dokku"]["ssh_keys"] = [{ user: "test", key: "public_key" }]
+        node.override["dokku"]["ssh_keys"] = [
+          { user: "test", key: "public_key" }
```
[the remark this answers is not in the document]
Comment (yourivdlans) [reply spec/unit/recipes/ssh_keys_spec.rb]: 😆

File: Gemfile [hunk 3]
```diff
@@ -1,8 +1,8 @@
 source "https://rubygems.org"
 
 group :integration do
-  gem "docker-api", "1.26.2"
-  gem "chef", "< 12.4"
-  gem "chefspec", "4.5.0"
+  gem "docker-api", "1.34.0"
+  gem "chef", "~> 14"
```
[the remark this answers is not in the document]
Comment (yourivdlans) [reply Gemfile]: You're right, sorry hound!

**Comment by yourivdlans** [comment]
Tests are green! I've also added the integration tests to travis.

Let me know what you think :)

File: .kitchen.docker.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+---
+driver:
+  name: docker
+  use_sudo: false
+  volume: /var/run/docker.sock:/var/run/docker.sock
+
+platforms:
+- name: ubuntu-14.04
```
Comment (nickcharlton) [inline .kitchen.docker.yml]: We mention 14.04 here, but 16.04 lower down (in `spec/spec_helper.rb`) and then trusty (14.04) in the `.travis.yml`. Which one should it be? Should we upgrade to 16.04 everywhere?

Comment (yourivdlans) [reply .kitchen.docker.yml]: Good point! I think we should upgrade to `16.04` yes.

**Comment by yourivdlans** [comment]
Its taken a while but I've upgraded the platform to 16.04 and tests are green!

**Pull request merged**